[new action] Linear actions

lindalala opened the pull request:
adds new actions and tests for Linear 
- getIssues with optional search query
- getProjects
- getIssueDetails
- getProjectDetails
- getTeams
- getTeamDetails

how to get a token
- follow https://linear.app/developers/oauth-2-0-authentication
- make sure you have admin scope

Test:
```
Test
lindali@Lindas-MacBook-Air actions-sdk % npm run test tests/linear/testGetTeams.ts


> @credal/actions@0.2.43 test
> node --loader ts-node/esm tests/linear/testGetTeams.ts

(node:91237) ExperimentalWarning: `--experimental-loader` may be removed in the future; instead use `register()`:
--import 'data:text/javascript,import { register } from "node:module"; import { pathToFileURL } from "node:url"; register("ts-node/esm", pathToFileURL("./"));'
(Use `node --trace-warnings ...` to show where the warning was created)
(node:91237) [DEP0180] DeprecationWarning: fs.Stats constructor is deprecated.
(Use `node --trace-deprecation ...` to show where the warning was created)
(node:91237) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
{
  "success": true,
  "teams": [
    {
      "id": "7a607ce0-900b-4d8b-a733-cf8c08f6f8f9",
      "name": "team10"
    },
    {
      "id": "cf8e7511-d4a2-49b0-b2be-4b631a1da388",
      "name": "Vulcan Collective"
    }
  ]
}
lindali@Lindas-MacBook-Air actions-sdk % npm run test tests/linear/testGetTeamDetails.ts


> @credal/actions@0.2.43 test
> node --loader ts-node/esm tests/linear/testGetTeamDetails.ts

(node:91662) ExperimentalWarning: `--experimental-loader` may be removed in the future; instead use `register()`:
--import 'data:text/javascript,import { register } from "node:module"; import { pathToFileURL } from "node:url"; register("ts-node/esm", pathToFileURL("./"));'
(Use `node --trace-warnings ...` to show where the warning was created)
(node:91662) [DEP0180] DeprecationWarning: fs.Stats constructor is deprecated.
(Use `node --trace-deprecation ...` to show where the warning was created)
(node:91662) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
{
  "success": true,
  "team": {
    "id": "cf8e7511-d4a2-49b0-b2be-4b631a1da388",
    "name": "Vulcan Collective",
    "identifier": "VUL",
    "members": [
      {
        "id": "157099b8-f55f-40fa-9557-0d16cd86d1c2",
        "name": "Linda Li",
        "email": "linda@vulcancollective.io"
      }
    ]
  }
}
lindali@Lindas-MacBook-Air actions-sdk % npm run test tests/linear/testGetProjects.ts


> @credal/actions@0.2.43 test
> node --loader ts-node/esm tests/linear/testGetProjects.ts

(node:97912) ExperimentalWarning: `--experimental-loader` may be removed in the future; instead use `register()`:
--import 'data:text/javascript,import { register } from "node:module"; import { pathToFileURL } from "node:url"; register("ts-node/esm", pathToFileURL("./"));'
(Use `node --trace-warnings ...` to show where the warning was created)
(node:97912) [DEP0180] DeprecationWarning: fs.Stats constructor is deprecated.
(Use `node --trace-deprecation ...` to show where the warning was created)
(node:97912) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
result:  {
  "success": true,
  "projects": [
    {
      "id": "f74d765b-0dc0-4bb7-8e0a-4995317f9180",
      "name": "Project 2",
      "status": "backlog",
      "labels": [],
      "creator": {
        "id": "157099b8-f55f-40fa-9557-0d16cd86d1c2",
        "name": "Linda Li"
      },
      "progress": 0,
      "url": "https://linear.app/vulcancollective/project/project-2-2a2e934372a3"
    },
    {
      "id": "642ece1c-aaa2-47a4-a7d3-e2bb9ab7475e",
      "name": "Test Project",
      "status": "backlog",
      "labels": [],
      "creator": {
        "id": "157099b8-f55f-40fa-9557-0d16cd86d1c2",
        "name": "Linda Li"
      },
      "lead": {
        "id": "157099b8-f55f-40fa-9557-0d16cd86d1c2",
        "name": "Linda Li"
      },
      "progress": 0,
      "url": "https://linear.app/vulcancollective/project/test-project-99839dd339df"
    }
  ]
}
lindali@Lindas-MacBook-Air actions-sdk % npm run test tests/linear/testGetProjectDetails.ts


> @credal/actions@0.2.43 test
> node --loader ts-node/esm tests/linear/testGetProjectDetails.ts

(node:99166) ExperimentalWarning: `--experimental-loader` may be removed in the future; instead use `register()`:
--import 'data:text/javascript,import { register } from "node:module"; import { pathToFileURL } from "node:url"; register("ts-node/esm", pathToFileURL("./"));'
(Use `node --trace-warnings ...` to show where the warning was created)
(node:99166) [DEP0180] DeprecationWarning: fs.Stats constructor is deprecated.
(Use `node --trace-deprecation ...` to show where the warning was created)
(node:99166) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
Result:  {
  "success": true,
  "project": {
    "id": "642ece1c-aaa2-47a4-a7d3-e2bb9ab7475e",
    "name": "Test Project",
    "description": "summary",
    "state": "started",
    "progress": 0,
    "createdAt": "2025-07-09T21:32:04.209Z",
    "updatedAt": "2025-07-11T22:13:35.121Z",
    "lead": {
      "id": "157099b8-f55f-40fa-9557-0d16cd86d1c2",
      "name": "Linda Li"
    },
    "team": {
      "id": "cf8e7511-d4a2-49b0-b2be-4b631a1da388",
      "name": "Vulcan Collective"
    },
    "issues": [
      {
        "id": "416715c7-0487-4c1a-b2a0-e2995dff5369",
        "name": "Test Issue #500",
        "url": "https://linear.app/vulcancollective/issue/VUL-513/test-issue-500"
      },
      {
        "id": "3e43c2fd-a740-4df3-a2bd-92a9c6dae37a",
        "name": "First issue",
        "url": "https://linear.app/vulcancollective/issue/VUL-10/first-issue"
      }
    ],
    "url": "https://linear.app/vulcancollective/project/test-project-99839dd339df",
    "updates": [
      {
        "id": "3949a133-dd0e-45c0-99e5-b15144212c8b",
        "content": "update message",
        "author_name": "Linda Li",
        "created_at": "2025-07-11T22:12:35.396Z"
      }
    ],
    "content": "summary"
  }
}
lindali@Lindas-MacBook-Air actions-sdk % npm run test tests/linear/testGetIssues.ts


> @credal/actions@0.2.43 test
> node --loader ts-node/esm tests/linear/testGetIssues.ts

(node:1090) ExperimentalWarning: `--experimental-loader` may be removed in the future; instead use `register()`:
--import 'data:text/javascript,import { register } from "node:module"; import { pathToFileURL } from "node:url"; register("ts-node/esm", pathToFileURL("./"));'
(Use `node --trace-warnings ...` to show where the warning was created)
(node:1090) [DEP0180] DeprecationWarning: fs.Stats constructor is deprecated.
(Use `node --trace-deprecation ...` to show where the warning was created)
(node:1090) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
{
  "success": true,
  "issue": {
    "id": "416715c7-0487-4c1a-b2a0-e2995dff5369",
    "title": "Test Issue #500",
    "description": "This is issue number 500",
    "state": "Backlog",
    "creator": {
      "id": "157099b8-f55f-40fa-9557-0d16cd86d1c2",
      "name": "Linda Li"
    },
    "team": {
      "id": "cf8e7511-d4a2-49b0-b2be-4b631a1da388",
      "name": "Vulcan Collective"
    },
    "project": {
      "id": "642ece1c-aaa2-47a4-a7d3-e2bb9ab7475e",
      "name": "Test Project"
    },
    "priority": 3,
    "dueDate": "2025-07-12",
    "createdAt": "2025-07-10T19:20:14.302Z",
    "updatedAt": "2025-07-11T22:24:04.305Z",
    "labels": [
      "three",
      "Bug"
    ],
    "url": "https://linear.app/vulcancollective/issue/VUL-513/test-issue-500",
    "comments": [
      {
        "author_name": "Linda Li",
        "comment": "???"
      },
      {
        "author_name": "Linda Li",
        "comment": "comment"
      }
    ],
    "content": "This is issue number 500"
  }
}
lindali@Lindas-MacBook-Air actions-sdk % npm run test tests/linear/testGetIssueDetails.ts


> @credal/actions@0.2.43 test
> node --loader ts-node/esm tests/linear/testGetIssueDetails.ts

(node:1533) ExperimentalWarning: `--experimental-loader` may be removed in the future; instead use `register()`:
--import 'data:text/javascript,import { register } from "node:module"; import { pathToFileURL } from "node:url"; register("ts-node/esm", pathToFileURL("./"));'
(Use `node --trace-warnings ...` to show where the warning was created)
(node:1533) [DEP0180] DeprecationWarning: fs.Stats constructor is deprecated.
(Use `node --trace-deprecation ...` to show where the warning was created)
(node:1533) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
{
  "success": true,
  "issue": {
    "id": "416715c7-0487-4c1a-b2a0-e2995dff5369",
    "title": "Test Issue #500",
    "description": "This is issue number 500",
    "state": "Backlog",
    "creator": {
      "id": "157099b8-f55f-40fa-9557-0d16cd86d1c2",
      "name": "Linda Li"
    },
    "team": {
      "id": "cf8e7511-d4a2-49b0-b2be-4b631a1da388",
      "name": "Vulcan Collective"
    },
    "project": {
      "id": "642ece1c-aaa2-47a4-a7d3-e2bb9ab7475e",
      "name": "Test Project"
    },
    "priority": 3,
    "createdAt": "2025-07-10T19:20:14.302Z",
    "updatedAt": "2025-07-11T22:04:28.246Z",
    "labels": [
      "three",
      "Bug"
    ],
    "url": "https://linear.app/vulcancollective/issue/VUL-513/test-issue-500",
    "comments": [
      {
        "author_name": "Linda Li",
        "comment": "???"
      },
      {
        "author_name": "Linda Li",
        "comment": "comment"
      }
    ],
    "content": "This is issue number 500"
  }
}
```

richrliu commented on src/actions/schema.yaml:
```diff
@@ -6729,3 +6729,271 @@ actions:
+              leadID:
+                type: string
+                description: ID of the project lead
+              teamID:
+                type: string
+                description: ID of the team the project belongs to
```
The docs say this might be multiple teams. Can we have this be `{id: str, name: str"`?

richrliu commented on src/actions/schema.yaml:
```diff
@@ -6729,3 +6729,271 @@ actions:
+                description: The issues in the project
+                items:
+                  id: 
+                    type: string
```
can we also include url, comments, content?

richrliu commented on src/actions/schema.yaml:
```diff
@@ -6729,3 +6729,271 @@ actions:
+              updatedAt:
+                type: string
+                description: When the project was last updated
+              leadID:
+                type: string
+                description: ID of the project lead
```
Can we have this be {id: str, name:str}

lindalala commented:
Updated! I saw a comment field on Project but it kept returning empty and I could not figure out how to comment on a project in the Linear UI. I found a project updates field instead that seemed reasonable and implemented that instead, but let me know if you actually want to grab the comments field
